content/doc/developer/internationalization/internationalization-and-localization.adoc

```diff
@@ -0,0 +1,301 @@
+|===
+a|
+....
+<f:entry title="${%title}" help="/plugin/myPlugin/help.html">
+....
+
+|===
+
+Then, just create the file help.html for English or help_xx.html for the
+'xx' language.
+
+== What translators need to know/do
+
+The Jenkins project always welcomes contributions to translations. If
+you are interested in helping us, just file a pull request on GitHub. In
+the remainder of this section, we discuss what needs to be translated
+and how.
+
+[cols="",]
+|===
+|Remember that properties files must be encoded in
+https://docs.oracle.com/javase/6/docs/api/java/util/Properties.html[ISO
+8859-1]. This should be the default anyway.
+|===
+
+=== Translation tools
+
+* https://wiki.jenkins.io/display/JENKINS/Translation-Tool[Translation
+Tool] : Command line tool to help translators to generate files, add
+missing keys, and more.
+* https://plugins.jenkins.io/translation/[Translation
+Assistance Plugin] : Adds a dialog box in Jenkins to translate and send
+missing keys.
+
+=== Translating `+Messages.properties+`
+
+Developers place messages that require localization in
+`+Messages.properties+`. Those need to be translated in the usual
+manner. See `+Messages_ja.properties+` in the core as an example if you
+are new to this process.
+
+If looking at `+Messages.properties+` alone does not give you
+enough contextual information about where the messages are used. you can access messages by with the
+type-safe `+Messages+` class generated by
+https://docs.oracle.com/middleware/1212/wls/WLAPI/weblogic/i18n/Localizer.html[Localizer]. To find out where
+messages are actually used, use your IDE to find all the usages of the
+message format method.
+
+=== Translating message references in Jelly
+
+The other messages that need to be translated are in Jelly view files,
+which are in `+src/main/resources/**.jelly+`. To localize them, first
+you run Maven to generate a skeleton property file for your locale:
+
+[width="100%",cols="100%",]
+|===
+a|
+....
+$ cd jenkins/core  (or a plugin dir)
+$ mvn stapler:i18n -Dlocale=fr
+....
+
+|===
+
+This generates `+*_fr.properties+` throughout
+`+src/main/resources+` with an empty value. If the file already exists,
+it appends missing entries to existing files.
+
+You can then translate messages in each property file.
+It is not necessary to translate the entire file;
+entries that are empty fall back to the default locale.
+
+[cols="",]
+|===
+|https://addons.mozilla.org/en-US/firefox/addon/locale-switcher/[Quick Locale Switcher]
+extension to firefox is useful to toggle between various locales.
+|===
+
+=== Translating static HTML resources
+
+Stand-alone HTML files are often used in Jenkins for things like inline
+help messages. To translate these resources, add the
+locale code between the file name and the extension. For example, the
+Japanese version of `+abc.html+` would be `+abc_ja.html+`, and the British
+version of it could be `+abc_en_GB.html+`. These files must be
+encoded in UTF-8.
```

Comment:
```suggestion
Stand-alone HTML files are often used in Jenkins for things like inline help messages.
To translate these resources, add the locale code between the file name and the extension.
For example, the Japanese version of `+abc.html+` would be `+abc_ja.html+`, and the British version of it could be `+abc_en_GB.html+`.
These files must be encoded in UTF-8.
```

Reply:
Done.